flask/app.py

```diff
@@ -1013,8 +1013,10 @@ def index():
 
         # starting with Flask 0.8 the view_func object can disable and
         # force-enable the automatic options handling.
-        provide_automatic_options = getattr(view_func,
-            'provide_automatic_options', None)
+        provide_automatic_options = options.pop(
+            'provide_automatic_options', getattr(view_func,
```

Comment:
I think it's better to add this as normal param to `add_url_rule` instead of popping from `options`, such that it can be documented in its docstring normally.